var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -378,6 +392,44 @@ def install(self, spec, prefix):
                 '--with-scalapack=0'
             ])
 
+        # For the moment, HPDDM does not work as a dependency
+        # using download instead
+        if '+hpddm' in spec:
+            options.extend(['--download-hpddm'])
+
+        if '+mmg' in spec:
+            options.extend([
+                '--with-mmg-dir=%s' %
+                spec['mmg'].prefix,
+                '--with-mmg=1'
+            ])
```

Comment:
Presumably this code can be re-formatted wrt changes in https://github.com/spack/spack/pull/25074 (so rebase after that one is merged?)

Reply:
Very well, I'll do so.

Reply:
#25074 is merged - and this can be rebsed. Here  - the code here would be: (if --with-pkg-include,--with-pkg-lib would work - its preferable)
```patch
diff --git a/var/spack/repos/builtin/packages/petsc/package.py b/var/spack/repos/builtin/packages/petsc/package.py
index 76d93ac424..0461e0e5da 100644
--- a/var/spack/repos/builtin/packages/petsc/package.py
+++ b/var/spack/repos/builtin/packages/petsc/package.py
@@ -417,6 +417,9 @@ def install(self, spec, prefix):
                 (jpeg_sp, 'libjpeg', True, True),
                 (scalapack_sp, 'scalapack', False, True),
                 'strumpack',
+                'mmg',
+                'parmmg',
+                'tetgen',
         ):
             # Cannot check `library in spec` because of transitive deps
             # Cannot check variants because parmetis keys on +metis
```


However - if only --with-pkg-dir works and not the above - use the format:

(`mmg, 'mmg', False, False),`

Reply:
I'm back, and changed the commits accordingly. I hope it fits the philosophy.
Tetgen failed with `'tetgen'` so I used `('tetgen', 'tetgen', False, False),`.

Reply:
Perhaps adding `include`, `libs` properties to tetgen/package.py  is a better fix?